### ISSUE TITLE
Accessibility - Generated DataGrid column headers do not contain role attribute with aria-label

### ISSUE THREAD

**Issue by Hannah-Frost** [mui/mui-x#4771]

**Comment by m4theushw** [comment]
I only found https://dequeuniversity.com/rules/axe/4.4/aria-allowed-attr?application=AxeChrome

The error makes sense. The div with the title doesn't have `role`, so `aria-label` is not allowed in it. We could fix this by wrapping the entire column header with the `Tooltip`. Since we don't have slots for headers I don't think it's a breaking change.
